### ISSUE TITLE
[FeatureRequest] reproducable build among npm and pnpm

### ISSUE THREAD

**Issue by loynoir** [evanw/esbuild#1499]
### Brief
When using package manager `pnpm`, compare with `npm`.
`esbuild --bundle` reproduce **different** output.

### `npm`
```js
// node_modules/fetch-blob/streams.cjs
var require_streams = __commonJS({
  "node_modules/fetch-blob/streams.cjs"() {
    // code
  }
});
```

### `pnpm`
```js
// node_modules/.pnpm/fetch-blob@3.1.2/node_modules/fetch-blob/streams.cjs
var require_streams = __commonJS({
  "node_modules/.pnpm/fetch-blob@3.1.2/node_modules/fetch-blob/streams.cjs"() {
    // code
  }
});
```

**Comment by loynoir** [comment]
pnpm: Fast, disk space efficient package manager

repo: [https://github.com/pnpm/pnpm](https://github.com/pnpm/pnpm)
install: [https://pnpm.io/installation](https://pnpm.io/installation)

**Comment by evanw** [comment]
The file path is the file path, and is not in esbuild's control. Non-minified output deliberately includes the file path because it's intended for debugging by a human. You will get different file paths with other different package managers too such as Yarn, which puts files inside zip files.

Relative file paths of input files are also included in the hashes present in output file names (a critical correctness feature for bugs such as #928), so even removing the file path comments and strings in the output files that you are pointing to does not guarantee reproducible builds.

Basically this is by design. Builds are only reproducible given the same file system contents. Use the same package manager if you want the same build output.

**Comment by hyrious** [comment]
You can use `--minify-identifiers` to remove the file path string in output.

**Comment by loynoir** [comment]
> You can use `--minify-identifiers` to remove the file path string in output.
@hyrious Seems not working on v0.12.17. I try `--minify-whitespace`, that works, thanks!

**Comment by loynoir** [comment]
@evanw Maybe provide `--minify-path`, for compactible build?

**Comment by evanw** [comment]
I'm closing this as "by design" as described above. The file structure is part of the input, and different inputs lead to different outputs.
